massa-execution-worker/src/execution.rs

```diff
@@ -1454,6 +1454,36 @@ impl ExecutionState {
                 // set the execution context
                 *context_guard!(self) = execution_context;
 
+                if req.fee.is_some() || req.coins.is_some() {
+                    let mut context = context_guard!(self);
+                    let call_stack_addr = context.get_call_stack();
+                    if let Some(fee) = req.fee {
+                        // Transfer fee from sender to none
+                        if let Some(addr) = call_stack_addr.get(0) {
+                            context.transfer_coins(
+                                Some(addr.clone()),
+                                None,
+                                Amount::from_raw(fee),
+                                false,
+                            )?;
+                        }
+                    }
+
+                    if let Some(coins) = req.coins {
+                        // Transfer coins from sender to called address
+                        if let Some(from) = call_stack_addr.get(0) {
+                            if let Some(to) = call_stack_addr.get(1) {
+                                context.transfer_coins(
+                                    Some(from.clone()),
+                                    Some(to.clone()),
+                                    Amount::from_raw(coins),
+                                    false,
+                                )?;
+                            }
+                        }
+                    }
+                }
+
                 // load and execute the compiled module
                 // IMPORTANT: do not keep a lock here as `run_function` uses the `get_module` interface
```

Comment:
We also need the `fee` parameter when doing a bytecode execution (but only function call has the coins parameter)

Reply:
Done